### PR TITLE
[PoC] Allow to uplod avatars via a public API

### LINKED ISSUE #10549: Add "public" OCS endpoint to manage the avatar
So our beautiful clients can manage the avatar too.

Currently it's only possible via `index.php/` which has CSRF protection that prevents mobile clients

### PR DESCRIPTION
**Pull request by rullzer**
Fixes #10549

* Avatar must be sqaure
* POST must be done using a mutlipart form
* Key is avatar

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

### CODE REVIEW

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 1]
```diff
@@ -984,4 +993,68 @@ public function resendWelcomeMessage(string $userId): DataResponse {
...
 	}
+
+	/**
+	 * @NoAdminRequired
+	 */
+	public function getAvatar(string $userId): DataResponse {
+		// TODO: Return link to avatar?
+	}
+
+	public function deleteAvatar(string $userId): DataResponse {
```
Comment (nickvergessen) [inline apps/provisioning_api/lib/Controller/UsersController.php]: `@NoAdminRequired`

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 2]
```diff
@@ -984,4 +993,68 @@ public function resendWelcomeMessage(string $userId): DataResponse {
...
+		$currentLoggedInUser = $this->userSession->getUser();
+
+		$targetUser = $this->userManager->get($userId);
+		if ($targetUser === null) {
```
Comment (nickvergessen) [inline apps/provisioning_api/lib/Controller/UsersController.php]: This should not be needed as you have to be logged in?

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 3]
```diff
@@ -984,4 +993,68 @@ public function resendWelcomeMessage(string $userId): DataResponse {
...
+		$avatar->remove();
+
+		return new DataResponse([]);
+	}
+
+	public function uploadAvatar(string $userId): DataResponse {
+		$currentLoggedInUser = $this->userSession->getUser();
+
+		$targetUser = $this->userManager->get($userId);
+		if ($targetUser === null) {
```
Comment (nickvergessen) [inline apps/provisioning_api/lib/Controller/UsersController.php]: see comments from above method

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 4]
```diff
@@ -984,4 +993,68 @@ public function resendWelcomeMessage(string $userId): DataResponse {
...
+		$subAdminManager = $this->groupManager->getSubAdmin();
+		if (!$subAdminManager->isUserAccessible($currentLoggedInUser, $targetUser)
+			&& !$this->groupManager->isAdmin($currentLoggedInUser->getUID())) {
+			// No rights
+			throw new OCSException('', \OCP\API::RESPOND_UNAUTHORISED);
+		}
+
+		$file = $this->request->getUploadedFile('avatar');
+
+		if ($file['size'] > 20 * 1024 * 1024) {
```
Comment (skjnldsv) [inline apps/provisioning_api/lib/Controller/UsersController.php:1044]: I would love to have this as a const somewhere :)

File: apps/provisioning_api/lib/Controller/UsersController.php [hunk 5]
```diff
@@ -984,4 +993,68 @@ public function resendWelcomeMessage(string $userId): DataResponse {
 
 		return new DataResponse();
 	}
+
+	/**
+	 * @NoAdminRequired
+	 */
+	public function getAvatar(string $userId): DataResponse {
+		// TODO: Return link to avatar?
```
Comment (skjnldsv) [inline apps/provisioning_api/lib/Controller/UsersController.php:1001]: yes! Cleaner :)
Optional parameter with size?

**Pull request closed without merging**